### ISSUE TITLE
Faster mesh intersections

### ISSUE THREAD

**Issue by BrunoLevy** [BrunoLevy/geogram#103]
- [x] `interval_nt` class in round_to_nearest mode
- [x]  `interval_nt` class in round_to_infinity mode (does not seem to be faster than round_to_nearest mode)
- [x] filters for `MeshInTriangle` predicates
- [x] filters for `RadialSort`
- [x] Do radial sort in parallel
- [ ] random shuffle for better radial sort load balancing ? -> won't do
- [x] filter stats for radial sort
- [x] radial sort by propagation
- [ ] `CDT2d` per-vertex predicate caching for constraint insertion -> won't do, not worth the additional complexity
- [x] Better management of predicate cache when inserting a duplicated vertex (begin_transaction() / end_transaction())
- [ ] Global vertices table: use symbolic information as vertex key instead of exact geometry -> won't do, does not seem to be a hot spot
- [ ] Use local meshes instead of global vertices table then merge them (by sorting the vertices) -> won't do for now

**Comment by Glizzy77** [comment]
Wack 💯💯
